src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/boutgifter/dto/BeregningsresultatBoutgifterDto.kt

```diff
@@ -40,15 +42,46 @@ private fun BeregningsresultatBoutgifter.filtrerFraOgMed(dato: LocalDate?): Bere
     return BeregningsresultatBoutgifter(perioder.filter { it.tom >= dato })
 }
 
-fun BeregningsresultatForLøpendeMåned.tilDto(): BeregningsresultatForPeriodeDto =
+private fun BeregningsresultatBoutgifter.skalBrukeDetaljertVisning(): Boolean =
+    perioder.any {
+        it.grunnlag.utgifter.keys
+            .contains(TypeBoutgift.UTGIFTER_OVERNATTING)
+    }
+
+fun BeregningsresultatForLøpendeMåned.tilDto(revurderFra: LocalDate?): BeregningsresultatForPeriodeDto =
     BeregningsresultatForPeriodeDto(
         fom = grunnlag.fom,
         tom = grunnlag.tom,
         stønadsbeløp = stønadsbeløp,
         utbetalingsdato = grunnlag.utbetalingsdato,
         sumUtgifter = summerUtgifter(),
-        utgifter = grunnlag.utgifter,
+        utgifter = finnUtgifterMedAndelTilUtbetaling(revurderFra),
         målgruppe = grunnlag.målgruppe,
         aktivitet = grunnlag.aktivitet,
         makssatsBekreftet = grunnlag.makssatsBekreftet,
     )
+
+fun BeregningsresultatForLøpendeMåned.finnUtgifterMedAndelTilUtbetaling(
+    revurderFra: LocalDate?,
+): List<UtgiftBoutgifterMedAndelTilUtbetaling> {
+    var totalSum = 0
+    return grunnlag.utgifter.values
+        .flatten()
+        .sorted()
+        .map { utgift ->
+            val skalUtbetales =
+                when {
+                    totalSum >= grunnlag.makssats -> 0
+                    totalSum + utgift.utgift <= grunnlag.makssats -> utgift.utgift
+                    else -> grunnlag.makssats - totalSum
+                }
```

Comment:
```suggestion
            val skalUtbetales = minOf(utgift, makssats - totalSum).coerceAtLeast(0)
```

Mulig dette kan være en funksjon i `Beregningsgrunnlag`? Som då kan unittestes, uten at man må forholde seg til hele listen i en test

Reply:
https://github.com/navikt/tilleggsstonader-sak/pull/709/commits/9e2e9d8f6be727b0385993836e4b5256a7b428f6